src/commander/mod.rs

```diff
@@ -170,9 +175,43 @@ impl Commander {
         Ok(())
     }
 
+    #[instrument(level = "trace", skip(self))]
     pub fn init(&self) -> Result<()> {
-        self.execute_void_jj_command(vec!["status"])
-            .context("Failed getting initial status")
+        // Ask jj about its version
+        let (color, quiet) = (false, false);
+        let found_version = self
+            .execute_jj_command(vec!["version"], color, quiet)
+            .context("Run jj version")?;
+
+        // Extract version number
+        if found_version[0..3] != *"jj " {
+            trace!("jj version output \"{}\"", found_version);
+            bail!("Jujutsu version string was not recognized");
+        }
+        let found_version = &found_version[3..].trim();
+
+        // Parse version numbers
+        use version_compare::Version;
+        let min_version =
+            Version::from(JJ_MIN_VERSION).expect("JJ_MIN_VERSION string should parse");
+        let found_version =
+            Version::from(found_version).expect("Output from jj version should parse");
+
+        trace!(
+            "Check jj version found ({}) >= min ({})",
+            found_version,
+            min_version
+        );
+
+        // Verify that jj is not too old
+        if found_version < min_version {
+            bail!(
+                "jujutsu version {} is too old. Must be at least {}",
+                found_version,
+                min_version
+            );
+        }
```

Comment:
This can be simplified
```suggestion
        trace!(found_version = found_version, min_version = min_version, "checking jj version");

        // Check when jj min_version > found_version
        if compare_to(min_version, found_version, Cmp::Gt).is_ok_and(|b| b) {
            bail!(
                "jujutsu version {} is too old. Must be at least {}",
                found_version,
                min_version
            );
        }
```

Reply:
Thank you. I will make a smaller version.